File: Problemset/shuffle-the-array/shuffle-the-array.cpp

// @Title: 重新排列数组 (Shuffle the Array)
// @Author: 846188037@qq.com
// @Date: 2020-07-15 20:22:34
// @Runtime: 8 ms
// @Memory: 9.6 MB

class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> newNums;
        for(int i=0;i<n;i++){
            newNums.push_back(nums[i]);
            newNums.push_back(nums[i+n]);

        }
        return newNums;

    }
};

File: Problemset/shuffle-the-array/shuffle-the-array.py

# @Title: 重新排列数组 (Shuffle the Array)
# @Author: 846188037@qq.com
# @Date: 2020-07-15 19:56:40
# @Runtime: 36 ms
# @Memory: 13.5 MB

class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        newNums=list()
        for i in range(n):
            newNums.extend((nums[i],nums[n+i]))
        return newNums
